OpenRA.Mods.Common/Traits/Repairable.cs

```diff
@@ -58,10 +60,22 @@ bool CanRepairAt(Actor target)
 			return self.Info.Traits.Get<RepairableInfo>().RepairBuildings.Contains(target.Info.Name);
 		}
 
+		bool CanRearmAt(Actor target)
+		{
+			return self.Info.Traits.Get<RepairableInfo>().RepairBuildings.Contains(target.Info.Name);
```

Comment:
You are getting the Info of the trait you are in. Just cache the info in the constructor.
Same for `CanRepairAt`